src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -1239,6 +1250,12 @@ await foreach (var item in ReadArrayAsync())
             }
         }
 
+        public JsonOperationContext GetContextForNewDocument()
+        {
+            _context.CachedProperties.NewDocument();
```

Comment:
this worries me, verify that we are doing that the same way that it was before refactoring